include/dpp/streams.h

```diff
@@ -0,0 +1,158 @@
+/************************************************************************************
+ *
+ * D++, A Lightweight C++ library for Discord
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2021 Craig Edwards and D++ contributors 
+ * (https://github.com/brainboxdotcc/DPP/graphs/contributors)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ************************************************************************************/
+#pragma once
+#include <dpp/export.h>
+#include <dpp/user.h>
+#include <dpp/channel.h>
+#include <dpp/cluster.h>
+#include <utility>
+
+
+namespace dpp {
+/**
+ * @brief Return end of message
+ * @return 0
+*/
+int end_msg();
+/**
+ * @brief Simple parent for streams
+ * 
+ */
+
+class DPP_EXPORT base_stream {
+    public:
+        /**
+         * @brief Build message in stream
+         * @param msg Text to add
+         * @return This stream, for purposes of chaining
+         */
+        base_stream& operator<<(const std::string& msg);
+        /**
+         * @brief Finish and send message in stream
+         * @param n Number (currently irrelevant)
+         * @return This stream, for purposes of chaining
+         */
+        base_stream& operator<<(const int& n);
+        //TODO Make action rows work with this
+        /**
+         * @brief Build message in stream
+         * @param c Component to add
+         * @return This stream, for purposes of chaining
+         */
+        base_stream& operator<<(const component &c);
+        /**
+         * @brief Build message in stream
+         * @param e Embed to add
+         * @return This stream, for purposes of chaining
+         */
+        base_stream& operator<<(const embed &e);
+        /**
+         * @brief Build message in stream
+         * @param f Pair of filename and filecontent to add
+         * @return This stream, for purposes of chaining
+         */
+        base_stream& operator<<(const std::pair<std::string,std::string> &f);
```

Comment:
You forgot the MIME type, there are 3 parameters to `dpp::message::add_file`.
I don't really like using pairs/tuples for this. Which field in the tuple is the filename, the filecontent? Why not create a type for it, with well-defined names?